### DISCUSSION TITLE
Project structure

### DISCUSSION THREAD

**Discussion by agtokty** [tarikguney/manager-automation-v2#2] [Ideas]
Hi @tarikguney 

First of all, this is great project to use daily work life like you stated for your problem. I just forked the repo and start review the code. I had never write azure function and bot software before, but I wondered and decided to follow up project. First thing that I realized is project namespace/folder structure. I think current structure makes it **harder to understand** what in where a little bit. 

here is my humble suggestions: 
* you may move google chat related actions under **/actions/google** folder, and when new integration(slack, ms teams) comes it can go under **/actions/slack**. 
*  gathering function implementations under a single namespace
*  declaring a Transform interface/abstract class and gather implementations under single namespace

I know, you may be aware of these things and change all structure based on new implementation already.

regards

**Comment by tarikguney** [comment]
I like the suggestions. Thanks.
